Improved Error struct

jeroenvervaeke opened the pull request:
I've done 2 things to make it easier to work with the `Error` struct:

1. Implement `Display` for `Error<E>` when `E` implements `Display`
2. Added a `std` feature which implements the `std::error::Error` for `Error<E>` if `E` implements it

By doing this you can print the error in `no_std` environments and use `?` in a `std` environment.

This PR does not introduce any breaking changes.

eldruin reviewed (commented):
Sounds good to me. 
Just a couple things before the merge:
- Add an entry about the addition to the changelog
- Add a simple test printing the errors

jeroenvervaeke commented:
I've added an entry to the changelog and I've also added a test for the `Display` implementation.
I did not add a test for the Error implementation because I'm not sure if you left out the `PartialEq` derive on `Error` on purpose.

I've also updated the workflow to run tests with and without the `std` feature enabled

eldruin reviewed (approved):
Looks good to me, thank you!
Adding `PartialEq` to the `Error` type should be fine. I will add it.
